composeapi.go

```diff
@@ -20,25 +20,45 @@ import (
 	"encoding/json"
 	"fmt"
 	"log"
+	"net/http"
+	"time"
 
 	"github.com/parnurzeal/gorequest"
 )
 
-var ()
-
 const (
 	apibase = "https://api.compose.io/2016-07/"
 )
 
 // Client is a structure that holds session information for the API
 type Client struct {
+	// The number of times to retry a failing request if the status code is
+	// retryable (e.g. for HTTP 429 or 500)
+	Retries int
+	// The interval to wait between retries. gorequest does not yet support
+	// exponential back-off on retries
+	RetryInterval time.Duration
+	// RetryStatusCodes is the list of status codes to retry for
+	RetryStatusCodes []int
+
 	apiToken string
 }
 
 // NewClient returns a Client for further interaction with the API
 func NewClient(apiToken string) (*Client, error) {
 	return &Client{
-		apiToken: apiToken,
+		apiToken:      apiToken,
+		Retries:       5,
+		RetryInterval: 3 * time.Second,
+		RetryStatusCodes: []int{
+			http.StatusRequestTimeout,
+			http.StatusTooManyRequests,
+			http.StatusInternalServerError,
+			http.StatusBadGateway,
+			http.StatusServiceUnavailable,
+			http.StatusGatewayTimeout,
+			http.StatusInsufficientStorage,
```

Comment:
You could retry `StatusInsufficientStorage`, but it's not likely to be resolved so expediently as to be useful in this context. Generally when Compose is out of capacity it requires the manual intervention of an engineer to add more hosts to the region/cluster.

Reply:
Sounds like it should be removed. Done.